### PR TITLE
Fix #3923 - default panel on pinned sv report

### LINKED ISSUE #3923: Panel default selection from large svs in cancer case report
2023-04-28 12:08:59 MacBook-Pro-3 scout.server.app[79614] ERROR Exception on /cust000/cancer_case/case_report [GET]
Traceback (most recent call last):
  File "/usr/local/miniconda3/envs/scout39n/lib/python3.9/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/miniconda3/envs/scout39n/lib/python3.9/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/miniconda3/envs/scout39n/lib/python3.9/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/miniconda3/envs/scout39n/lib/python3.9/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/miniconda3/envs/scout39n/lib/python3.9/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/dannil/sandbox/scout/scout/server/utils.py", line 85, in decorated_function
    context = func(*args, **kwargs)
  File "/Users/dannil/sandbox/scout/scout/server/blueprints/cases/views.py", line 219, in case_report
    data = controllers.case_report_content(store, institute_id, case_name)
  File "/Users/dannil/sandbox/scout/scout/server/blueprints/cases/controllers.py", line 611, in case_report_content
    panel_ids=[
  File "/Users/dannil/sandbox/scout/scout/server/blueprints/cases/controllers.py", line 614, in <listcomp>
    if panel.is_default is True
AttributeError: 'dict' object has no attribute 'is_default'

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data
https://scout-stage.scilifelab.se/cust000/balsamic-validation-gmsmyeloid-tumor-only-2/cancer/sv-variants

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 - Freeze of werkzeug library to version<2.3 to avoid problems resulting from the consequential upgrade of the Flask lib
 - Huge list of genes in case report for megabases-long structural variants.
 - Fix displaying institutes without associated cases on institutes page
+- Fix default panel selection on pinned SVs in cancer case report
```
Comment (northwestwitch) [inline CHANGELOG.md:23]:
```suggestion
- Fix default panel selection on SVs in cancer case report
```

**Comment by dnil** [comment]
Before:
Screenshot 2023-04-28 at 12 21 26

After:
Screenshot 2023-04-28 at 12 23 46

File: scout/server/blueprints/cases/controllers.py [hunk 2]
```diff
@@ -611,7 +611,7 @@ def case_report_content(store, institute_id, case_name):
             panel_ids=[
                 panel["panel_id"]
                 for panel in case_obj.get("panels", [])
-                if panel.is_default is True
+                if panel.get("is_default", False) is True
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:614]: 👍🏻

**Review by northwestwitch — APPROVED** [review]
I approve 😄

**Comment by northwestwitch** [comment]
Cool, so this is working now, we can check also the relative checkbox in the new release PR!

**Pull request merged**